### ISSUE TITLE
Ability to view Helm Charts deployed by YAML manifest with helm CLI

### ISSUE THREAD

**Issue by dnoland1** [rancher/k3s#1143]
**Is your feature request related to a problem? Please describe.**
Yes, cannot use the helm CLI to view Helm Charts that were deployed using a HelmChart manifest. For example, if I put this file in /var/lib/rancher/k3s/server/manifests -
```yaml
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: cert-manager
  namespace: kube-system
spec:
  chart: cert-manager
  repo: https://charts.jetstack.io
  targetNamespace: cert-manager
  version: v0.9.1
```
It will deploy the cert-manager helm chart. But if I run this in the helm CLI, I don't see it:
```
root@ip-10-0-11-20:~# helm list -a --all-namespaces
NAME                	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART          	APP VERSION
root@ip-10-0-11-20:~#
```
Tested with helm 3.0.0:
```
root@ip-10-0-11-20:~# helm version
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

**Describe the solution you'd like**
Would be nice to if you can run helm list and helm status commands against k3s when deployed with a k3s helm chart manifest. Might also be nice to include the helm binary in a similar fashion to how kubectl and crictl are included.

**Comment by rancher-max** [comment]
I've verified this using k3s v1.18.9+k3s1. I used both the helm chart above and another random one:
```
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: grafana
  namespace: kube-system
spec:
  chart: stable/grafana
  targetNamespace: monitoring
  set:
    adminPassword: "NotVerySafePassword"
  valuesContent: |-
    image:
      tag: master
    env:
      GF_EXPLORE_ENABLED: true
    adminUser: admin
    sidecar:
      datasources:
        enabled: true
```
Neither show up using `helm list -a -A`. The chart for traefik does show up, however, as this is one of the default charts with k3s. More details of commands run and their results below:
```
$ k get pods -A
NAMESPACE     NAME                                     READY   STATUS              RESTARTS   AGE
kube-system   coredns-7944c66d8d-5dmzs                 0/1     Running             0          22s
kube-system   helm-install-grafana-88sxl               1/1     Running             0          22s
kube-system   local-path-provisioner-6d59f47c7-jk8js   1/1     Running             0          22s
kube-system   metrics-server-7566d596c8-qnpln          1/1     Running             0          22s
kube-system   traefik-758cd5fc85-zhn4b                 0/1     ContainerCreating   0          2s
kube-system   svclb-traefik-b85xx                      0/2     ContainerCreating   0          2s
kube-system   helm-install-traefik-d96gc               0/1     Completed           0          22s

$ helm list -a -A
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /etc/rancher/k3s/k3s.yaml
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /etc/rancher/k3s/k3s.yaml
NAME   	NAMESPACE  	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
traefik	kube-system	1       	2020-10-14 15:13:44.820636261 +0000 UTC	deployed	traefik-1.81.0	1.7.19 
```

**Comment by brandond** [comment]
I don't know how Helm finds things to display in the list, but the packaged Traefik chart and the user chart from your example end up deployed the exact same way.

@rancher-max I see that your helm-install-grafana pod is still Running, not Completed. That means the Helm chart isn't done installing yet. Can you give it a little longer, and confirm your `helm list` output AFTER the install pod is done running?

**Comment by dnoland1** [comment]
Helm looks for owner=helm label on secrets when running a `list`:
```
# helm list -a -A -v=6
I1014 23:35:36.234691  706174 loader.go:375] Config loaded from file:  /etc/rancher/k3s/k3s.yaml
I1014 23:35:36.239524  706174 loader.go:375] Config loaded from file:  /etc/rancher/k3s/k3s.yaml
I1014 23:35:36.257633  706174 round_trippers.go:443] GET https://127.0.0.1:6443/version?timeout=32s 200 OK in 3 milliseconds
I1014 23:35:36.258304  706174 loader.go:375] Config loaded from file:  /etc/rancher/k3s/k3s.yaml
I1014 23:35:36.275689  706174 round_trippers.go:443] GET https://127.0.0.1:6443/api/v1/secrets?labelSelector=owner%3Dhelm 200 OK in 16 milliseconds
```
The kubectl equivalent of the GET https - `kubectl get secrets -A -l owner=helm`

**Comment by rancher-max** [comment]
This looks to be working in both k3s v1.18.9 and the default installation of v1.19.13 in both centos and ubuntu. I also tried helm v3.0.0 and the current available version via snap of v3.4.1.

I installed certmanager chart using:
```yaml
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: cert-manager
  namespace: kube-system
spec:
  chart: cert-manager
  repo: https://charts.jetstack.io
  targetNamespace: cert-manager
  version: v1.1.0
```
I had to also run `kubectl create namespace cert-manager` for this after installing k3s, but then it installed successfully and was available to helm:
```
$ helm list -a --all-namespaces
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /etc/rancher/k3s/k3s.yaml
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /etc/rancher/k3s/k3s.yaml
NAME        	NAMESPACE   	REVISION	UPDATED                                	STATUS  	CHART              	APP VERSION
cert-manager	cert-manager	1       	2020-11-30 17:45:03.328403456 +0000 UTC	deployed	cert-manager-v1.1.0	v1.1.0     
traefik     	kube-system 	1       	2020-11-30 17:44:08.584097278 +0000 UTC	deployed	traefik-1.81.0     	1.7.19    
```

I'm going to close this out for now, but if it's noticed again, maybe it's something to do with versions of the charts deployed and not getting the secret label correctly populated as mentioned above.
